Docker and Kubernetes layers fixes

robbyoconnor opened the pull request:
- fix kubernetes-tramp deprecation warning by setting a threshold to exclude
- adjust docker-tramp min version threshold too

lebensterben commented:
any explanation?

robbyoconnor commented:
It's deprecated. Load the kubernetes layer on Emacs 29.1 or greater.

lebensterben commented:
reference?

robbyoconnor commented:
As I said, run it under emacs 29.1+

lebensterben commented:
You expect me to write this in the commit message?

"Run it in Emacs 29.1"?

robbyoconnor commented:
@lebensterben this is the warning:

> Warning (emacs): Package ‘kubernetes-tramp’ has been obsoleted, please use integrated package ‘tramp-container’

lebensterben commented on layers/+tools/kubernetes/packages.el:
```diff
@@ -27,7 +27,7 @@
     kubernetes
     kubernetes-evil
-    kubernetes-tramp))
+    (kubernetes-tramp :toggle (version< emacs-version "29.1"))))
```
```suggestion
    (kubernetes-tramp :toggle (version< emacs-version "29.0"))))
```

there is no Emacs 29.1 yet.

robbyoconnor replied:
I'll fix it. Same fix was done for docker-tramp so just used that one.

robbyoconnor replied:
Some people run master, myself included.

robbyoconnor replied:
Actually this also catches 30+ which is what emacs master is at.

lebensterben replied:
see https://github.com/emacs-mirror/emacs/tree/a259d0dda3878a64373b808627c4b4cab3971194

it's been there since 29.0.50

robbyoconnor replied:
This still fixes the problem.

robbyoconnor replied:
This still fixes the problem. 

What I'm saying is that this catches Emacs 30+

robbyoconnor replied:
Okay here's how to reproduce it: Run emacs snapshot and have the kubernetes layer. 

Let's pick a version threshold. Sorry for coming off brash but there was a very similar fix committed to the docker layer.

robbyoconnor replied:
https://github.com/syl20bnr/spacemacs/commit/50825419ab86983e0019c1142d548b0389c50c01

robbyoconnor replied:
Here's the PR you approved: https://github.com/syl20bnr/spacemacs/pull/15899

lebensterben replied:
I will fix that.

robbyoconnor replied:
What value should I put? 29.0?

robbyoconnor replied:
I'll do it: what value should I use?

robbyoconnor replied:
Actually: 

See: https://www.emacswiki.org/emacs/TrampAndDocker

> NOTE: if running Emacs built from the sources (or v29.1), as of Nov-2022 (probably earlier) there’s a native support for Docker via tramp (tramp-container.el) so the above package is not required.

lebensterben replied:
> I'll do it: what value should I use?

29.0.50

Just use git blame to find where the warning is added, then browse files at that point. You will see the README containing 29.0.50.

The EmacsWiki page is erroneous.

robbyoconnor replied:
Okay .. I'll fix docker-tramp for you. 

It will happen tomorrow my time (NY time)

robbyoconnor commented:
Updates made @lebensterben -- sorry for arguing with you.

Commit:
Docker and Kubernetes layers fixes

- fix kubernetes-tramp deprecation warning by setting a threshold to exclude
- adjust docker-tramp min version threshold too

robbyoconnor commented:
I adjusted everything @lebensterben

lebensterben commented:
thx

robbyoconnor commented:
Not a problem. I feel bad that I was so abrupt @lebensterben
